modules/ppcp-wc-gateway/src/Settings/SettingsListener.php

```diff
@@ -208,6 +208,25 @@ function () use ( $exception ) {
 		exit;
 	}
 
+	/**
+	 * Ensure 3DS contingency use `SCA_ALWAYS` instead of `3D_SECURE`.
+	 *
+	 * @return void
+	 * @throws NotFoundException When a setting was not found.
+	 */
+	public function listen_for_3d_secure_contingency(): void {
+		if ( ! $this->is_valid_site_request() || $this->settings->get( '3d_secure_contingency' ) !== '3D_SECURE' ) {
+			return;
+		}
+
+		$this->settings->set( '3d_secure_contingency', 'SCA_ALWAYS' );
+		$this->settings->persist();
+
+		$redirect_url = admin_url( 'admin.php?page=wc-settings&tab=checkout&section=ppcp-gateway' );
```

Comment:
Maybe should redirect to the cards tab? `admin.php?page=wc-settings&tab=checkout&section=ppcp-gateway&ppcp-tab=ppcp-credit-card-gateway`

Or maybe redirect not needed here at all.

Also as I understand it is executed only if the merchant saves settings. Maybe better to do that simply on some kind of plugin init/upgrade, such as `woocommerce_paypal_payments_gateway_migrate`.

Reply:
yup, much better on plugin upgrade, thanks!